kanidmd/src/lib/utils.rs

```diff
@@ -37,6 +38,14 @@ pub fn password_from_random() -> String {
     rand_string
 }
 
+pub fn backup_code_from_random() -> HashSet<String> {
+    let mut set = std::collections::HashSet::with_capacity(8);
```

Comment:
Minor, but can't you do:

```
(1..8).iter().map(|_| {
    readable_password_from_random()
})
.collect()
```

There is some type-hint magic in rust that pre-hints the capacity to the hashset during the collect so you don't need to do it manually.